protocols/gossipsub/src/behaviour.rs

```diff
@@ -3226,7 +3220,8 @@ where
                     for (raw_message, validation_error) in invalid_messages {
                         self.handle_invalid_message(
                             &propagation_source,
-                            &raw_message,
+                            &raw_message.topic,
+                            None,
```

Comment:
I find this funky tbh, with this we have `handle_invalid_message` defined with an argument that we know specifically when it's going to be none. Looking into this I wonder if we need `handle_invalid_message` at all and can just repeat the  required code, wdyt?

Reply:
I believe the compiler can figure this out. Current implementation makes it less verbose.

Reply:
Friendly ping @jxs